syncapi/synctypes/clientevent.go

```diff
@@ -363,14 +363,17 @@ func updatePowerLevelEvent(userIDForSender spec.UserIDForSender, se gomatrixserv
 		}
 		newPls[user] = level
 	}
-	var newPlBytes, newEv []byte
-	newPlBytes, err = json.Marshal(newPls)
-	if err != nil {
-		return nil, err
-	}
-	newEv, err = sjson.SetRawBytes(se.JSON(), "content.users", newPlBytes)
-	if err != nil {
-		return nil, err
+	var newPlBytes []byte
+	newEv := se.JSON()
+	if len(pls.Users) > 0 {
```

Comment:
I think you mean nil check here? Else it's impossible to set `users: {}`? Not that I'm sure why you'd want to..

Reply:
For the particular test where there was no users field, I wanted to not present an empty users field to the client. With our logic we were adding in the field even if it wasn't present in the first place.
I forgot that someone may actually want to have an empty users field.

I've changed the logic now to allow an empty users field while also not adding it in if it isn't there in the first place.